contents/javascript/Javascript-framework-Redux.md

```diff
@@ -0,0 +1,231 @@
+
+## Why Use Redux?
+
+<box type='definition' header="Some points given in this section were adapted from these articles">
+
+* [Why use Redux? Reasons with clear examples by Neo Ighodaro](https://blog.logrocket.com/why-use-redux-reasons-with-clear-examples-d21bffd5835/)
+* [Redux · An Introduction by Alex Bachuk](https://www.smashingmagazine.com/2016/06/an-introduction-to-redux/)
+</box>
+
+Now that we know what Redux is, let us take a look at how the structure of Redux and its concepts make possible some benefits.
+
+1. **State is made predictable**<br/>
+In Redux, the single immutable store is the single source of truth for application state. Neither the view of the application, nor network callbacks directly write to it. Changes to the state are via actions, which Redux centralizes and keep in strict order. Also, reducers are pure functions, always producing the same result for the same state and action. Because of all of this, there is a structured and predictable cycle of which applicable state is updated, making state is predictable. Having a predictable state, makes it possible to implement functionality that would be otherwise difficult, such as Undo and Redo.
+
+2. **Easier to debug and inspect**<br/>
+Since actions and state are plain objects, it is possible to serialize them and log them. In Redux, especially with the [Redux DevTools](https://github.com/reduxjs/redux-devtools), you can view a log of actions that took place, making it easy to understand errors in code. You can also print the store <tooltip content="for example using `console.log(store.getState())`">within the code itself</tooltip>. The Redux DevTools, used in the browser, also provides features such as a tree visualization of the application state, as well a time travel ability to replay state changes.
+
+![Redux Devtool](https://user-images.githubusercontent.com/7957859/48663602-3aac4900-ea9b-11e8-921f-97059cbb599c.png)
+
+<center>
+
+_Redux DevTools_ <sup>[image source](https://user-images.githubusercontent.com/7957859/48663602-3aac4900-ea9b-11e8-921f-97059cbb599c.png)</sup>
+</center>
+
+3. **Easier testing**<br/>
+Using Redux makes for testable code. Since the reducers used to change the state are pure functions, it is easy make tests. Writing small, pure and isolated reducer functions in Redux, means we write small and isolated tests. 
+
+4. **Allows for server-side rendering**<br/>
```

Comment:
I would personally replace this with a simpler and more key point like asynchronous actions ( since there are hardly applications without this ), but if you're keeping to this, some comments:

`Allows` -> not definitive, switch to a softer word
Explain the contrary first, how it is difficult to achieve SSR without a state management framework, then explain how redux makes this easier.